docs/tools/terraform/howto/vnet-peering-azure.rst

```diff
@@ -0,0 +1,215 @@
+    resource "azuread_application_password" "app_password" {
+      application_object_id = azuread_application.app.object_id
+    }
+
+    # 5. Find the id properties of your virtual network
+    # Skip, we have values in the state
+
+    # 6. Grant your service principal permissions to peer
+    resource "azurerm_role_assignment" "app_role" {
+      role_definition_name = "Network Contributor"
+      principal_id         = azuread_service_principal.app_principal.object_id
+      scope                = azurerm_virtual_network.virtual_network.id
+    }
+
+    # 7. Create a service principal for the Aiven application object
+    # Yes, application_id is hardcoded.
+    resource "azuread_service_principal" "aiven_app_principal" {
+      application_id = "55f300d4-fc50-4c5e-9222-e90a6e2187fb"
+    }
+
+    # 8. Create a custom role for the Aiven application object
+    resource "azurerm_role_definition" "role_definition" {
+      name        = "my-azure-role-definition"
+      description = "Allows creating a peering to vnets in scope (but not from)"
+      scope       = "/subscriptions/${data.azurerm_subscription.subscription.subscription_id}"
+
+      permissions {
+        actions = ["Microsoft.Network/virtualNetworks/peer/action"]
+      }
+
+      assignable_scopes = [
+        "/subscriptions/${data.azurerm_subscription.subscription.subscription_id}"
+      ]
+    }
+
+    # 9. Assign the custom role to the Aiven service principal
+    resource "azurerm_role_assignment" "aiven_role_assignment" {
+      role_definition_id = azurerm_role_definition.role_definition.role_definition_resource_id
+      principal_id       = azuread_service_principal.aiven_app_principal.object_id
+      scope              = azurerm_virtual_network.virtual_network.id
+
+      depends_on = [
+        azuread_service_principal.aiven_app_principal,
+        azurerm_role_assignment.app_role
+      ]
+    }
+
+    # 10. Find your AD tenant id
+    # Skip, it's in the env
+
+    # 11. Create a peering connection from the Aiven Project VPC
+    # 12. Wait for the Aiven platform to set up the connection
+    resource "aiven_azure_vpc_peering_connection" "peering_connection" {
+      vpc_id                = aiven_project_vpc.avn_vpc.id
+      peer_resource_group   = azurerm_resource_group.resource_group.name
+      azure_subscription_id = data.azurerm_subscription.subscription.subscription_id
+      vnet_name             = azurerm_virtual_network.virtual_network.name
+      peer_azure_app_id     = azuread_application.app.application_id
+      peer_azure_tenant_id  = "00000000-0000-0000-0000-000000000000"
+
+      depends_on = [
+        azurerm_role_assignment.aiven_role_assignment
+      ]
+    }
+
+
+Step 2: Create peering in Azure
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Now create the connection using the credentials from the previous step.
+Unfortunately terraform doesn't support `dynamic provider configuration <https://github.com/hashicorp/terraform/issues/25244>`_.
```

Comment:
```suggestion
.. note::

Terraform doesn't support dynamic provider configuration.
```

Reply:
Thanks for the clarification on this! I've made a small adjustment to the language and removed the link (so that we don't have to update it later).